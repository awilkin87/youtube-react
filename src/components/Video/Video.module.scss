.videoContainer {
  position: relative;
  width: 100%;

  &:before {
    content: "";
    display: block;
    padding-top: 56.25%; // 9:16 display ratio
  }

  .video {
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;

    .videoPlayer {
      min-height: 240px;
      width: 100%;
      height: 100%;
    }
  }
}