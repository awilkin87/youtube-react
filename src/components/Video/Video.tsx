import styles from './Video.module.scss';

const BASE_EMBED_URL = 'https://www.youtube.com/embed/';

export interface VideoProps {
  id: string
}

const Video: React.FunctionComponent<Readonly<VideoProps>> = function (props) {
  // TODO Enable autoplaying after development
  const embedUrl = `${BASE_EMBED_URL}${props.id}?autoplay=1&modestbranding=1`;
  return (
    <div className={styles.videoContainer}>
      <div className={styles.video}>
        <iframe className={styles.videoPlayer}
          title='video'
          src={embedUrl}
          frameBorder='0'
          allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture'
          allowFullScreen />
      </div>
    </div>
  );
}

export default Video;