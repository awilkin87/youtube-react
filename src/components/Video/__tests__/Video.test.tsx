import { shallow } from "enzyme";
import Video from '../Video';

test('Renders VideoPreview', () => {
  const wrapper = shallow(<Video id='Test123' />);
  expect(wrapper).toMatchSnapshot();
});