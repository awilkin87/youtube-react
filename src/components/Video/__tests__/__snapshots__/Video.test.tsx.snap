// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders VideoPreview 1`] = `
<div
  className="videoContainer"
>
  <div
    className="video"
  >
    <iframe
      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      className="videoPlayer"
      frameBorder="0"
      src="https://www.youtube.com/embed/Test123?autoplay=0&modestbranding=1"
      title="video"
    />
  </div>
</div>
`;
